cilium/cmd/debuginfo.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	pkg "github.com/cilium/cilium/pkg/client"
+
+	"github.com/spf13/cobra"
+)
+
+// debuginfoCmd retrieves debugging related information from the daemon
+var debuginfoCmd = &cobra.Command{
+	Use:   "debuginfo",
+	Short: "Display available debugging information of daemon",
+	Run: func(cmd *cobra.Command, args []string) {
+		debugInfoDaemon()
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(debuginfoCmd)
+	// TODO: add option for generating HTML file
+}
+
+func debugInfoDaemon() {
+	if resp, err := client.Daemon.GetDebuginfo(nil); err != nil {
+		fmt.Fprintf(os.Stderr, "%s\n", pkg.Hint(err))
+		os.Exit(1)
+	} else {
```

Comment:
You can avoid this else branch as you will quit the program above.

Reply:
Sure, was copy pasta.